src/stores.rs

```diff
@@ -607,6 +607,196 @@ impl<K: Hash + Eq, V> Cached<K, V> for TimedCache<K, V> {
     }
 }
 
+/// Timed LRU Cache
+///
+/// Stores a limited number of values,
+/// evicting expired and least-used entries.
+///
+/// Note: This cache is in-memory only
+#[derive(Clone, Debug)]
+pub struct TimedSizedCache<K, V> {
+    store: HashMap<K, (Instant, usize)>,
+    order: LRUList<(K, V)>,
+    capacity: usize,
+    seconds: u64,
+    hits: u64,
+    misses: u64,
+}
+
+impl<K: Hash + Eq, V> TimedSizedCache<K, V> {
+    /// Creates a new `SizedCache` with a given size limit and pre-allocated backing data
+    pub fn with_size_and_lifespan(size: usize, seconds: u64) -> TimedSizedCache<K, V> {
+        if size == 0 {
+            panic!("`size` of `TimedSizedCache` must be greater than zero.")
+        }
+        TimedSizedCache {
+            store: HashMap::with_capacity(size),
+            order: LRUList::<(K, V)>::with_capacity(size),
+            capacity: size,
+            seconds,
+            hits: 0,
+            misses: 0,
+        }
+    }
+
+    // TODO: Is it desired behavior to list expired elements in ordered lists?
```

Comment:
I think we should expire items when providing iterators over keys and values. Returning expired keys/values seems odd since you would get back `None` if you were to do a `.cache_get` right after